python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java

```diff
@@ -451,10 +453,32 @@ public void visitAssignmentStatement(AssignmentStatement assignmentStatement) {
     getFirstAssignmentName(assignmentStatement).ifPresent(lhsName -> {
       var assignedValueType = assignmentStatement.assignedValue().typeV2();
       lhsName.typeV2(assignedValueType);
-      addStaticFieldToClass(lhsName);
+      addStaticFieldToClass(lhsName, PythonType.UNKNOWN);
     });
   }
 
+  @Override
+  public void visitAnnotatedAssignment(AnnotatedAssignment assignmentStatement) {
+    scan(assignmentStatement.assignedValue());
+    scan(assignmentStatement.annotation());
+    scan(assignmentStatement.variable());
```

Comment:
any reason you don't call `super`? Tests in `TypeInferenceV2Test` are still green if I use the call to `super`
```suggestion
    super.visitAnnotatedAssignment(assignmentStatement);
```